Libs/DICOM/Core/ctkDICOMDatabase.cpp

```diff
@@ -803,11 +801,16 @@ bool ctkDICOMDatabasePrivate::insertPatientStudySeries(const ctkDICOMItem& datas
   else
   {
     logger.debug("Insert new patient if not already in database: " + patientID + " " + patientsName);
-    if (this->insertPatient(dataset, patientID, patientsName, dbPatientID))
+    int patientMetadataInsertSuccess = this->insertPatient(dataset, patientID, patientsName, dbPatientID);
```

Comment:
"success" is a `bool` type, not `int`. The `-1` and `1` act like magic numbers here. Two cleaner options could be:
- Option A: `int insertedPatientsCount = this->insertPatient...` and then `insertedPatientsCount>0`.
- Option B: use an enum to distinguish between inserted/not-inserted/failed.
- Option C: add additional arguments that can return more detail.